### ISSUE TITLE
DB/Quest: A Fine Mess (Gnomeregan) - NPC does not exit instance (despawn)

### ISSUE THREAD

**Issue by TQPS** [TrinityCore/TrinityCore#25857]
**Description:**

The quest titled [A Fine Mess](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?quest=2904) found/started inside Gnomeregan cannot be completed because the NPC [Kernobee](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=7850) does not exit instance (despawn).

<details>

NPC stops short of instance exit and remains there.

![image](https://user-images.githubusercontent.com/66294837/104793312-0efd8880-57f6-11eb-9c92-5164c4a7a6ed.png)

You can exit the instance yourself, come back in and NPC is still waiting by the exit.

You can continue with the dungeon and come back to find NPC is still waiting by the exit.

![image](https://user-images.githubusercontent.com/66294837/104793373-584dd800-57f6-11eb-8b37-004af8c3fcdd.png)

</details>

**Expected behaviour:**

NPC should exit instance (despawn) per action list

https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=7850#sai-actionlist-785000

<details>

Notice that action #3 & #4 still has not executed either at this point.

![image](https://user-images.githubusercontent.com/66294837/104793507-09ed0900-57f7-11eb-81d6-5e5f8f570bef.png)

</details>

**Steps to reproduce the problem:**

1. Enter Gnomeregan and head to the area where Kernobee is
<details>

![image](https://user-images.githubusercontent.com/66294837/104793697-e8405180-57f7-11eb-897c-1fb8495637f6.png)

</details>

2. Accept his quest "A Fine Mess" and escort him to the exit as instructed.
3. NPC waits at exit and does not despawn

**Branch(es):**
3.3.5

TC rev. hash/commit:
TrinityCore rev. f6b22df9bd22 2021-01-13 00:53:43 +0200 (3.3.5 branch) (Win64, RelWithDebInfo, Static)

**Operating system:**
Windows 7 Ultimate x64 SP1

**Comment by Jinnaix** [comment]
Was working as expected in my test.

Note to myself: Use english client for tests with screenshots xD
![WoWScrnShot_011621_101006](https://user-images.githubusercontent.com/37972361/104807956-0d769500-57e3-11eb-845a-d144ceb4ff5e.jpg)

**Comment by illfated** [comment]
Completed correctly for me about 2-3 weeks ago. First time I tried, the quest failed because I got lost and forgot which way is out from that room. :P
Admittedly, main characters are on `account_access`.`SecurityLevel`=3 (GM-level 3/admin), so even if I play with GM off + GM visible on, I think it does not count as 100% correct for testing. Also, when I ran around there, just to get the quests done, my character was level 50+, so even if I instigated fights with some NPCs on the way out, it did not hinder me much. I was almost surprised to see Kernobee despawn after a little while and the quest completion show up on my screen. Maybe I should test it from the Alliance side.

**Comment by TQPS** [comment]
I will retry this quest again when I go in next; perhaps it just glitched for some reason?

On a side note - have you noticed if you die inside Gnomeregan you spawn all the way at Kharanos? I'm certain if you died in Gnomeregan in WOTLK you spawned at the graveyard in Iceflow Lake - I cant find anything to support this.

**Comment by Jinnaix** [comment]
@illfated - in general, the account security level doesnt change anything.
There are some scripts in the core that are written for testing encounters and check your account for CanBeGameMaster()
![image](https://user-images.githubusercontent.com/37972361/104808530-2da85300-57e7-11eb-93f8-15442268d172.png)

 And in most cases its totally fine to check stuff with gm on/off. But there are scripts in the core that behave different or wont run if gm is on, areatrigger in most cases:
![image](https://user-images.githubusercontent.com/37972361/104808580-92fc4400-57e7-11eb-8e6c-5b0af3940e9b.png)

**Comment by illfated** [comment]
> On a side note - have you noticed if you die inside Gnomeregan you spawn all the way at Kharanos? I'm certain if you died in Gnomeregan in WOTLK you spawned at the graveyard in Iceflow Lake - I cant find anything to support this.

Have not checked nor tested this, but maybe you are right. Isn't there a graveyard even closer to the Gnomeregan entrance?

**Comment by illfated** [comment]
> @illfated - in general, the account security level doesnt change anything.
> There are some scripts in the core that are written for testing encounters and check your account for CanBeGameMaster()
> 
> And in most cases its totally fine to check stuff with gm on/off. But there are scripts in the core that behave different or wont run if gm is on, areatrigger in most cases:

Thank you, very good indeed. Nice to know and to be reminded of those features.

**Comment by TQPS** [comment]
> > On a side note - have you noticed if you die inside Gnomeregan you spawn all the way at Kharanos? I'm certain if you died in Gnomeregan in WOTLK you spawned at the graveyard in Iceflow Lake - I cant find anything to support this.
> 
> Have not checked nor tested this, but maybe you are right. Isn't there a graveyard even closer to the Gnomeregan entrance?

Yes, that's at the lake - Iceflow Lake. Kharanos is a half zone hike back to the instance. I have been trying to find confirmation of the proper graveyard but it seems no one has ever died in Gnomeregan. Been reluctant to raise it without supportive research.

**Comment by TQPS** [comment]
Ran the instance again - no NPC to be found. Checked all the rooms. Had Low Level Quests enabled as well.

It's as though the quest thinks it's finished but it never updated.

Remains incompletable.

<details>

![image](https://user-images.githubusercontent.com/66294837/104810074-82c78180-5846-11eb-84a0-383a3875eed9.png)

</details>

**Comment by illfated** [comment]
That is sort of weird, but I presume you have to drop the quest to make Kernobee spawn again. When the quest failed the first time for me due to time-out, I had to abandon the quest and retake it when Kernobee finally respawned. The second time it completed correctly, although it took some seconds for the completion message to appear on my screen. It is possible that some of my game client addons could have caused the extra delay.

**Comment by Jinnaix** [comment]
Drop the Quest, otherwise you cant try it again.
NPC has a lot of possible spawns, you will only find him in one of these locations:

![image](https://user-images.githubusercontent.com/37972361/104810515-5c2d2a80-57f5-11eb-96de-823cd1ce5edd.png)

**Comment by dr-j** [comment]
Think you need to triggger area trigger 1105 while you have kernobee and alarm o bot following you.  Areatrigger sets data 2 2 to alarm bot

**Comment by TQPS** [comment]
> Think you need to triggger area trigger 1105 while you have kernobee and alarm o bot following you. Areatrigger sets data 2 2 to alarm bot

I think you're right - the Alarm-o-bot did not follow us to exit yesterday. I ran the quest again today and this time the bot followed - this did allow the quest to complete. 

<details>

![image](https://user-images.githubusercontent.com/66294837/104830130-ed70cf80-58cf-11eb-869c-358c2c8709e1.png)

</details>

**Comment by Faq** [comment]
Kernobee should continue to the exit even without alarm-o-bot or despawn when bot not in some range?

**Comment by illfated** [comment]
As mentioned above by dr-j , it may not work on TC unless **Alarm-a-bomb 2600** is present, due to the scripting.
- https://wow.gamepedia.com/Alarm-a-bomb_2600
- https://www.wowhead.com/npc=7897/alarm-a-bomb-2600#comments
- https://wow.gamepedia.com/A_Fine_Mess
- https://www.wowhead.com/quest=2904/a-fine-mess
- https://classic.wowhead.com/quest=2904/a-fine-mess#comments
